build.gradle

```diff
@@ -1,139 +1,122 @@
     }
 }
-plugins {
-    id 'com.github.johnrengelman.shadow' version '2.0.4'
-    id 'java'
-}
+
+// Apply plugins
 apply plugin: 'net.minecraftforge.gradle.forge'
 apply plugin: 'org.spongepowered.mixin'
+apply plugin: 'eclipse'
 
+// Set basic properties
+version = project.version
+group = project.group
+archivesBaseName = project.artifact
 
-version = "Alpha9"
-group = "com.minecrafttas" // http://maven.apache.org/guides/mini/guide-naming-conventions.html
-archivesBaseName = "TASmod-1.12.2"
-
-sourceCompatibility = targetCompatibility = '1.8' // Need this here so eclipse task generates correctly.
-compileJava {
-    sourceCompatibility = targetCompatibility = '1.8'
-}
+// Compile for java 8
+sourceCompatibility = targetCompatibility = JavaVersion.VERSION_1_8
 
+// Configure forge
 minecraft {
-    version = "1.12.2-14.23.5.2847"
-    runDir = "run"
+	version = forge_version
+	runDir = "run"
+	mappings = mappings_version
     
-    mappings = "snapshot_20171003"
-    makeObfSourceJar = false // an Srg named sources jar is made by default. uncomment this to disable.
-    replace '${version}':project.version,'${mcversion}':project.minecraft.version
     clientRunArgs += "--mixin mixins.tasmod.json --tweakClass org.spongepowered.asm.launch.MixinTweaker"
     serverRunArgs += "--mixin mixins.tasmod.json --tweakClass org.spongepowered.asm.launch.MixinTweaker"
 }
 
+// Dependency repositories
 repositories {
     mavenCentral()
+    maven { url = "https://jitpack.io" }
     maven { url = "https://maven.mgnet.work/main" }
-    maven { url 'https://jitpack.io' }
-    maven {
-        name = 'sponge'
-        url = 'https://repo.spongepowered.org/maven'
-    }
+    maven { url = 'https://repo.spongepowered.org/maven' }
 }
 
+// Dependency configurations
 configurations {
-    downloadMod{
+    downloadMod {
     	transitive = false
     }
+    embed
+    compile.extendsFrom(embed)
 }
 
+// Dependencies
 dependencies {
-    compile('org.spongepowered:mixin:0.8.2') {
-        exclude module: 'launchwrapper'
-        exclude module: 'guava'
-        exclude module: 'gson'
+	// tasmod dependencies (exported with jar)
+    embed group: 'com.dselent', name: 'bigarraylist', version: '1.0'
+    embed('org.spongepowered:mixin:0.8.2') {
+    	exclude module: 'launchwrapper'
+    	exclude module: 'log4j-core'
         exclude module: 'commons-io'
-        exclude module: 'log4j-core'
+        exclude module: 'gson'
+        exclude module: 'guava'
     }
-    compile 'org.msgpack:msgpack-core:0.8.16'
-    compile 'org.msgpack:jackson-dataformat-msgpack:0.8.16'
-    //compile files('libs/KillTheRNG-1.12.2-2.0-WIP-deobf.jar')
-    compile 'com.minecrafttas:killtherng:2.0-SNAPSHOT'
-    //compile files('libs/BigArrayList-1.3.jar')
-    compile 'com.dselent:bigarraylist:1.0'
     
+    // tasmod dependencies (only for compilation)
+    compileOnly group: 'com.minecrafttas', name: 'killtherng', version: '2.0-SNAPSHOT'
```

Comment:
Would've changed it on my own, but I can't edit the changes for some reason... Please switch this out to `killtherng, version 2.0` instead of the snapshot version, same with download killtherng

Reply:
done